NewsApp-main/NewsApp/Scenes/NewsDetail/ViewController/NewsDetailViewController.swift

```diff
@@ -0,0 +1,370 @@
+        viewModel = NewsDetailViewModel(service: networkService)
+        viewModel.view = self
+        viewModel.viewWillAppear()
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configure()
+    }
+
+    override var shouldAutorotate: Bool {
+      return true
+    }
+
+    override var supportedInterfaceOrientations: UIInterfaceOrientationMask {
+      return [UIInterfaceOrientationMask.landscapeLeft,
+              UIInterfaceOrientationMask.landscapeRight,
+              UIInterfaceOrientationMask.portrait ];
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        stopAutoScrolling()
+    }
+
+    // MARK: - UI Configure
+    private func configure() {
+        title = sourceTitle
+        view.backgroundColor = .white
+        view.addSubview(scrollView)
+        scrollView.addSubview(contentView)
+        contentView.addSubview(sliderCollectionView)
+        contentView.addSubview(newsListCollectionView)
+        contentView.addSubview(activityIndicator)
+        contentView.addSubview(pageController)
+
+        scrollView.refreshControl = refreshControl
+        refreshControl.addTarget(self, action: #selector(refreshData), for: .valueChanged)
+
+        // MARK: Constraints
+
+        scrollView.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
+
+        contentView.snp.makeConstraints { (make) in
+            make.top.bottom.equalTo(self.scrollView)
+            make.left.right.equalTo(self.view)
+            make.width.equalTo(self.scrollView)
+            make.height.equalTo(self.scrollView)
+        }
+        sliderCollectionView.snp.makeConstraints { make in
+            make.top.equalToSuperview().offset(16)
+            make.height.equalToSuperview().multipliedBy(0.40)
+            make.left.equalToSuperview().offset(16)
+            make.right.equalToSuperview().offset(-16)
+        }
+
+        newsListCollectionView.snp.makeConstraints { make in
+            make.top.equalTo(pageController.snp.bottom).offset(8)
+            make.bottom.equalToSuperview().offset(8)
+            make.left.equalToSuperview().offset(16)
+            make.right.equalToSuperview().offset(-16)
+        }
+
+        activityIndicator.snp.makeConstraints { make in
+            make.center.equalToSuperview()
+        }
+
+        pageController.snp.makeConstraints { make in
+            make.top.equalTo(sliderCollectionView.snp.bottom).offset(-32)
+            make.centerX.equalToSuperview()
+            make.width.equalToSuperview().multipliedBy(0.45)
+            make.height.equalToSuperview().multipliedBy(0.05)
+        }
+
+        startAutoScrolling()
+        setupAutoRefreshTimer()
+    }
+
+    // MARK: Start scroll slide
+    func startAutoScrolling() {
+        slideTimer = Timer.scheduledTimer(timeInterval: 5.0, target: self, selector: #selector(autoScroll), userInfo: nil, repeats: true)
```

Comment:
Bu logic timer vs ViewModel'da olabilir. Test edebilmek için orada kullanabilirsin. Timer'ı mocklar test sırasında saniye bekletmeden fire fonksiyonunun çağırılmasını ve sağlayabilirsin. Veya aynı sürede teste bir delay koyabilirsin. biraz araştır çok kolay bulunur. View ve controller'da business logic sayısı 0 olmalı